Modules/Quest/QuestgiverFrame.lua

```diff
@@ -0,0 +1,147 @@
+---@class QuestgiverFrame
+local QuestgiverFrame = QuestieLoader:CreateModule("QuestgiverFrame")
+
+---@type QuestieDB
+local QuestieDB = QuestieLoader:ImportModule("QuestieDB")
+
+local _G = _G
+local tinsert = tinsert
+local MAX_NUM_QUESTS = MAX_NUM_QUESTS
+
+local titleLines = {}
+local questIconTextures = {}
+
+-- This is the logic used for determining which icon we should show for a quest
+---@param questID number
+---@param isActive boolean
+---@return string
+local function determineAppropriateQuestIcon(questID, isActive)
+    local icon = Questie.icons["available"] -- fallback icon in case any of the logic below fails
+    if isActive == true then
+        icon = Questie.icons["incomplete"] -- fallback icon in case any of the logic below fails
+        if QuestieDB.IsComplete(questID) == 1 then
+            if QuestieDB.IsPvPQuest(questID) then
+                icon = Questie.icons["pvpquest_complete"]
+            elseif QuestieDB.IsActiveEventQuest(questID) then
+                icon = Questie.icons["eventquest_complete"]
+            elseif QuestieDB.IsRepeatable(questID) then
+                icon = Questie.icons["repeatable_complete"]
+            else
+                icon = Questie.icons["complete"]
+            end
+        end
+    else
+        if QuestieDB.IsPvPQuest(questID) then
+            icon = Questie.icons["pvpquest"]
+        elseif QuestieDB.IsActiveEventQuest(questID) then
+            icon = Questie.icons["eventquest"]
+        elseif QuestieDB.IsRepeatable(questID) then
+            icon = Questie.icons["repeatable"]
+        end
+    end
+    return icon
+end
+
+-- 9.0.0 API GOSSIP
+local function updateGossipFrame()
+    local numAvailable = GetNumGossipAvailableQuests()
+    local numActive = GetNumGossipActiveQuests()
+    local availQuests = {GetGossipAvailableQuests()}
```

Comment:
```suggestion
    local availQuests = {QuestieCompat.GetAvailableQuests()}
```

Reply:
Using `QuestieCompat` here will help us in the future, when we extend `QuestieCompat` to be a good mapping layer returning a modern structure and not that index based stuff from the old API. Even though this code can't be reached on the DF API (yet).